intel_adsp: sim: several logging test build failures

### Describe the bug


First observed in https://github.com/zephyrproject-rtos/zephyr/actions/runs/18811754455.

Several logging tests have build failures on the Intel ADSP simulated platforms because my change to `soc/intel/intel_adsp/Kconfig.defconfig` disables WINSTREAM_CONSOLE if SIMULATOR_XTENSA, as the ADSP logging backend uses the winstream trace out:

```
subsys/logging/backends/log_backend_adsp.c:29: undefined reference to `winstream_console_trace_out'
```

Regression caused by b3a7eeaa7925fdde203b6df1843a7984579ea731.

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -p -b intel_adsp/ace15_mtpm/sim tests/subsys/logging/log_core_additional -T logging.async
```

### Relevant log output

```shell
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/laurenmu/zephyrproject/zephyr/build/zephyr/zephyr_pre0.map 
: && ccache /home/laurenmu/zephyr-sdk-0.17.4/xtensa-intel_ace15_mtpm_zephyr-elf/bin/xtensa-intel_ace15_mtpm_zephyr-elf-gcc  -gdwarf-4 -Os zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/xtensa/core/offsets/offsets.c.obj  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map,/home/laurenmu/zephyrproject/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/xtensa/core/libarch__xtensa__core.a  zephyr/arch/arch/xtensa/core/startup/libarch__xtensa__core__startup.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/lib/posix/c_lib_ext/liblib__posix__c_lib_ext.a  zephyr/soc/soc/ace15_mtpm/common/libintel_adsp_common.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/crypto/libdrivers__crypto.a  zephyr/drivers/mm/libdrivers__mm.a  zephyr/drivers/power_domain/libdrivers__power_domain.a  zephyr/drivers/timer/libdrivers__timer.a  modules/xtensa/libmodules_xtensa_hal.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L/home/laurenmu/zephyrproject/zephyr/build/zephyr  zephyr/arch/common/libisr_tables.a  zephyr/soc/soc/ace15_mtpm/common/libintel_adsp_common.a  -fuse-ld=bfd  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -Wl,--undefined=_sw_isr_table  -Wl,--undefined=_irq_vector_table  -specs=picolibc.specs  -DPICOLIBC_DOUBLE_PRINTF_SCANF -L/home/laurenmu/zephyr-sdk-0.17.4/xtensa-intel_ace15_mtpm_zephyr-elf/bin/../lib/gcc/xtensa-intel_ace15_mtpm_zephyr-elf/12.2.0 -lc -lgcc && cd /home/laurenmu/zephyrproject/zephyr/build/zephyr && /usr/bin/cmake -E true
/home/laurenmu/zephyr-sdk-0.17.4/xtensa-intel_ace15_mtpm_zephyr-elf/bin/../lib/gcc/xtensa-intel_ace15_mtpm_zephyr-elf/12.2.0/../../../../xtensa-intel_ace15_mtpm_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(log_backend_adsp.c.obj):(.literal.char_out+0x0): undefined reference to `winstream_console_trace_out'
/home/laurenmu/zephyr-sdk-0.17.4/xtensa-intel_ace15_mtpm_zephyr-elf/bin/../lib/gcc/xtensa-intel_ace15_mtpm_zephyr-elf/12.2.0/../../../../xtensa-intel_ace15_mtpm_zephyr-elf/bin/ld.bfd: zephyr/libzephyr.a(log_backend_adsp.c.obj): in function `char_out':
/home/laurenmu/zephyrproject/zephyr/subsys/logging/backends/log_backend_adsp.c:29: undefined reference to `winstream_console_trace_out'
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: Linux
- Toolchain: Zephyr SDK
-  f0cbba0f8ef502adeee7b784577b9994a5e9e8a6

### Additional Context

_No response_

--- a/soc/intel/intel_adsp/ace/Kconfig.defconfig.series
+++ b/soc/intel/intel_adsp/ace/Kconfig.defconfig.series
@@ -52,8 +52,11 @@ config DYNAMIC_INTERRUPTS
 
 if LOG
 
+config LOG_BACKEND_XTENSA_SIM
+	default y if SIMULATOR_XTENSA
+
 config LOG_BACKEND_ADSP
-	default y
+	default y if !SIMULATOR_XTENSA
 
 endif # LOG
 

